themes/bootstrap3/templates/search/facet-list.phtml

```diff
@@ -28,36 +26,45 @@
     <label><?=$this->translate('Sort') ?></label>
     <div class="btn-group">
       <?php foreach ($this->sortOptions as $key => $sort): ?>
-        <a href="<?=$urlBase . '&amp;facetpage=1&amp;facetsort=' . urlencode($key) ?>" class="btn btn-default js-facet-sort<?php if ($this->sort == $key): ?> active<?php endif; ?>" data-sort="<?=$key ?>" data-lightbox-ignore><?=$this->translate($sort) ?></a>
+        <a href="<?=$urlBase . '&facetpage=1&facetsort=' . urlencode($key) . '&contains=' . urlencode($contains) ?>" class="btn btn-default js-facet-sort<?php if($this->sort == $key): ?> active<?php endif; ?>" data-sort="<?=$key ?>"><?=$this->translate($sort) ?></a>
       <?php endforeach; ?>
     </div>
+
+    <?php if (method_exists($this->results->getParams(), 'setFacetContains')): ?>
+      <label><?=$this->transEsc('Filter') ?></label>
+      <div class="btn-group">
+        <form>
+          <input class="ajax_param" data-name="contains" class="form-control" type="text" value="<?=htmlspecialchars($this->contains)?>">
+          <button id="btn-reset-contains" type="button" role="button"><?=$this->icon('trash')?></button>
```

Comment:
This probably needs an aria-label or some other way of making it more accessible, since it currently doesn't seem to include a textual representation of its meaning. Maybe the reset button in the search box could be used as a model.

Reply:
I added the aria-labels as in the searchbox. Do you think it's ok like this or is it necessary to use different texts? I also left the icon as it was before, however i noticed that the searchbox uses a special ui-reset-search icon that we might be able to use here as well it you think that looks better.